components/supervisor/pkg/terminal/terminal.go

```diff
@@ -77,20 +78,23 @@ func (m *Mux) Start(cmd *exec.Cmd, options TermOptions) (alias string, err error
 	go func() {
 		term.waitErr = cmd.Wait()
 		close(term.waitDone)
-		_ = m.CloseTerminal(alias, 0*time.Second)
+
+		m.mu.Lock()
+		defer m.mu.Unlock()
+		_ = m.doClose(alias, unix.SIGKILL)
 	}()
 
 	return alias, nil
 }
 
-// Close closes all terminals with closeTerminaldefaultGracePeriod.
+// Close closes all terminals
 func (m *Mux) Close() error {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
 	var err error
 	for k := range m.terms {
-		cerr := m.doClose(k, closeTerminaldefaultGracePeriod)
+		cerr := m.doClose(k, unix.SIGTERM)
```

Comment:
Would be good to run this in parallel for all terminals. So they all get max time to stop.

Reply:
I understood that we should send SIGHUP instead of SIGTERM because that is what bash would forward to their processes.

Reply:
I also was thinking about it, but was not sure. From other side VS Code does it always like that when killing terminals. I would assume if you run a script which can be in executed in regular VS Code and super terminal then it should respect SIGHUP.